pkg/provider/gitlab/gitlab.go

```diff
@@ -45,6 +45,13 @@ type Provider struct {
 }
 
 func (v *Provider) Validate(ctx context.Context, params *params.Run, event *info.Event) error {
+	if event.Provider.WebhookSecret == "" {
```

Comment:
if user didn't set a webhook secret but there is a secret coming from github we should fail,  not allow it :)